### PR TITLE
[CBRD-24835] Added reading the schema information of the target object that created the synonym

### PR DESCRIPTION
**Pull request by airnet73**
http://jira.cubrid.org/browse/CBRD-24835

**Purpose**
Synonyms were added in cubrid 11.2. However, there is no way to read synonym schema information through cci api.
As a way to read synonym information is needed, a way to read schema information of the user who created the synonym using cci_schema_info() is needed.

If the user name and synonym name are entered in cci_schema_info(), the schema information of the target object must be read when the synonym is created.

In this issue, only the function to read the column information of the target object corresponding to the synonym was added.

**Implementation**
To read the column information of the target object in the synonym, the type below must be used.
Type : CCI_SCH_ATTR_WITH_SYNONYM

If CCI_SCH_ATTR_WITH_SYNONYMand [schema].[synonym] are used for the type and class_name parameters of the cci_schema_info() function, column information of the synonym target object is returned.

cci_schema_info(conn, CCI_SCH_ATTR_WITH_SYNONYM, "u1.s1", NULL, CCI_ATTR_NAME_PATTERN_MATCH, &error);

**Remarks**
N/A

### CODE REVIEW

File: src/broker/broker_cas_cci.h [hunk 1]
```diff
@@ -192,7 +192,8 @@ typedef enum
   CCI_SCH_IMPORTED_KEYS,
   CCI_SCH_EXPORTED_KEYS,
   CCI_SCH_CROSS_REFERENCE,
-  CCI_SCH_LAST = CCI_SCH_CROSS_REFERENCE
+  CCI_SCH_ATTR_INCLUDE_SYNONYM,
+  CCI_SCH_LAST = CCI_SCH_ATTR_INCLUDE_SYNONYM
```
Comment (kisoo-han) [inline src/broker/broker_cas_cci.h]: You should also reflect this modification to **cas_cci.h** of cubrid-cci repo.

Comment (airnet73) [reply src/broker/broker_cas_cci.h]: All right.
I will PR the cubrid-cci repo.

File: src/broker/cas_execute.c [hunk 2]
```diff
@@ -289,6 +289,8 @@ static int sch_class_info (T_NET_BUF * net_buf, char *class_name, char pattern_f
 			   T_BROKER_VERSION client_version);
 static int sch_attr_info (T_NET_BUF * net_buf, char *class_name, char *attr_name, char pattern_flag, char flag,
 			  T_SRV_HANDLE *);
+static int sch_attr_include_synonym_info (T_NET_BUF * net_buf, char *class_name, char *attr_name, char pattern_flag,
```
Comment (beyondykk9) [inline src/broker/cas_execute.c]:
```suggestion
static int sch_attr_with_synonym_info (T_NET_BUF * net_buf, char *class_name, char *attr_name, char pattern_flag,
```

File: src/broker/cas_execute.c [hunk 3]
```diff
@@ -3978,6 +3981,9 @@ ux_schema_info (int schema_type, char *arg1, char *arg2, char flag, T_NET_BUF *
     case CCI_SCH_CLASS_ATTRIBUTE:
       err_code = sch_attr_info (net_buf, arg1, arg2, flag, 1, srv_handle);
       break;
+    case CCI_SCH_ATTR_INCLUDE_SYNONYM:
```
Comment (beyondykk9) [inline src/broker/cas_execute.c]:
```suggestion
    case CCI_SCH_ATTR_WITH_SYNONYM:
```

File: src/broker/cas_execute.c [hunk 4]
```diff
@@ -3978,6 +3981,9 @@ ux_schema_info (int schema_type, char *arg1, char *arg2, char flag, T_NET_BUF *
...
+      err_code = sch_attr_include_synonym_info (net_buf, arg1, arg2, flag, 0, srv_handle);
```
Comment (beyondykk9) [inline src/broker/cas_execute.c]:
```suggestion
      err_code = sch_attr_with_synonym_info (net_buf, arg1, arg2, flag, 0, srv_handle);
```

File: src/broker/cas_execute.c [hunk 5]
```diff
@@ -4032,7 +4038,8 @@ ux_schema_info (int schema_type, char *arg1, char *arg2, char flag, T_NET_BUF *
 
   if (schema_type == CCI_SCH_CLASS || schema_type == CCI_SCH_VCLASS || schema_type == CCI_SCH_ATTRIBUTE
       || schema_type == CCI_SCH_CLASS_ATTRIBUTE || schema_type == CCI_SCH_QUERY_SPEC
-      || schema_type == CCI_SCH_DIRECT_SUPER_CLASS || schema_type == CCI_SCH_PRIMARY_KEY)
+      || schema_type == CCI_SCH_DIRECT_SUPER_CLASS || schema_type == CCI_SCH_PRIMARY_KEY
+      || schema_type == CCI_SCH_ATTR_INCLUDE_SYNONYM)
```
Comment (beyondykk9) [inline src/broker/cas_execute.c]:
```suggestion
      || schema_type == CCI_SCH_ATTR_WITH_SYNONYM)
```

File: src/broker/cas_execute.c [hunk 6]
```diff
@@ -8526,6 +8533,247 @@ sch_attr_info (T_NET_BUF * net_buf, char *class_name, char *attr_name, char patt
   return 0;
 }
 
+static int
+sch_attr_include_synonym_info (T_NET_BUF * net_buf, char *class_name, char *attr_name, char pattern_flag,
```
Comment (beyondykk9) [inline src/broker/cas_execute.c]:
```suggestion
sch_attr_with_synonym_info (T_NET_BUF * net_buf, char *class_name, char *attr_name, char pattern_flag,
```

File: src/broker/broker_cas_cci.h [hunk 7]
```diff
@@ -192,7 +192,8 @@ typedef enum
   CCI_SCH_IMPORTED_KEYS,
   CCI_SCH_EXPORTED_KEYS,
   CCI_SCH_CROSS_REFERENCE,
-  CCI_SCH_LAST = CCI_SCH_CROSS_REFERENCE
+  CCI_SCH_ATTR_INCLUDE_SYNONYM,
```
Comment (beyondykk9) [inline src/broker/broker_cas_cci.h]:
```suggestion
  CCI_SCH_ATTR_WITH_SYNONYM,
```

**Comment by airnet73** [comment]
@kisoo-han @beyondykk9 
Changed function name and type name.
Please confirm.

**Review by kisoo-han — APPROVED** [review]

**Pull request merged**